### ISSUE TITLE
Launching app with #1032 pops up permission screen in summary

### ISSUE THREAD

**Issue by shankari** [e-mission/e-mission-docs#998]
Steps to reproduce:
- native build with https://github.com/e-mission/e-mission-phone/pull/1032 included
- launch app (on android)
- sign in with a staging token
- the summary page (looking much uglier than before since it is all stuck to the top of the screen) shows up
- almost immediately, the permission popup is displayed and nothing else can be done

Please see screenshot below showing the permission popup over the summary screen

| Popup screen blocking summary | optimization is with the poup now! | After resolving, uglier summary screen with text all scrunched up |
|-----------|------------|---------------|
| <img width="418" alt="Screenshot 2023-10-02 at 11 44 22 PM" src="https://github.com/e-mission/e-mission-docs/assets/2423263/7fd13b95-425b-4e13-802e-3eb95b683eec"> | <img width="418" alt="Screenshot 2023-10-02 at 11 47 42 PM" src="https://github.com/e-mission/e-mission-docs/assets/2423263/90ce31ed-d54b-4b62-8b97-0c7d0e54e02d"> | <img width="418" alt="Screenshot 2023-10-02 at 11 48 07 PM" src="https://github.com/e-mission/e-mission-docs/assets/2423263/12a0b784-da52-4f5b-bb29-3259a72a8368"> |

@JGreenlee @Abby-Wheelis

**Comment by shankari** [comment]
Also, I am not sure that backwards compat works properly. If I launch an existing installed app that hasn't gone through the onboarding process, I get
<img width="418" alt="Screenshot 2023-10-02 at 11 54 22 PM" src="https://github.com/e-mission/e-mission-docs/assets/2423263/313573e1-7990-4258-a47f-28264495f527">

**Comment by JGreenlee** [comment]
I will attempt to reproduce and fix ASAP

**Comment by JGreenlee** [comment]
The first issue was a regression from the `summary` page being added. The onboarding flow wasn't originally designed with the summary being its own page, and I forgot to add it to the list of checks for whether the AppStatusModal should be allowed to show or not.
The popup should show after the `consent` page. To ensure this I was just checking that the route was not `welcome` or `consent`.

While fixing this, I'm going to refactor and have the onboarding state be an ordered `enum` (a TypeScript feature - nonexistent in plain JS). This way, we can say something like `onboardingRoute > Routes.CONSENT` to check whether we are past the consent page or not, instead of having to specifically list routes .

**Comment by JGreenlee** [comment]
For the ugly-looking summary page, I'm not sure what is desired. In my opinion, there is just not enough content there to warrant a full page - which was part of the reason I suggested combining it with the privacy policy.
Ideally, we'd have some graphics content to show around the text and take up the empty space. It currently looks decent on smaller screens, but large screens have too much space for this little bit of text.

To take up space, we could increase the margins and padding and we could make the text bigger. But we can't make phones smaller.

**Comment by shankari** [comment]
@JGreenlee not a showstopper for staging, but a reminder to test out the "install over existing app" (left screenshot) from
https://github.com/e-mission/e-mission-docs/issues/998#issuecomment-1744317295

wrt ugly-looking summary page, that is not a showstopper for staging

I think that what we had last time was indeed a much bigger font and a lot more padding. We could also add a logo (not sure if we did or not), and potentially even have the logo be one that the partner provides. However, that is not a showstopper to get it onto staging - you can brainstorm with the team at the meeting tomorrow and make changes.

The goal is to have a simple, visually appealing screen that introduces participants to the app and the program before throwing a bunch of legal jargon at them.  The equivalent of a "Welcome" or "Hola" screen while installing a phone, for example.

I can merge the permission screen fix now if it is ready and you can address the others later today.

**Comment by JGreenlee** [comment]
The permission popup fix is ready and I'll look at the others later today.

**Comment by JGreenlee** [comment]
<img width="771" alt="image" src="https://github.com/e-mission/e-mission-docs/assets/15843932/0f7e44c4-d09a-4e47-a8e6-a0a0270bcd0f">

Does this look better? (on one big screen and one small screen)

**Comment by shankari** [comment]
Sure, that is not ugly any more. I still don't think it looks as beautiful as the current welcome screen.

Note that this is also a welcome screen of sorts - not a welcome to the app, but a welcome to the study/program. But I'm fine with merging this and seeing if you want to improve it later.

**Comment by shankari** [comment]
@JGreenlee  https://github.com/e-mission/e-mission-docs/issues/998#issuecomment-1744317295 is still pending, would you like me to file a new issue?

**Comment by JGreenlee** [comment]
> @JGreenlee  https://github.com/e-mission/e-mission-docs/issues/998#issuecomment-1744317295 is still pending, would you like me to file a new issue?

I am not sure whether the backwards-compat issue will have already been resolved by the other fixes. I am unsure what specific steps I would take to reproduce it.

Can you clarify what state the old install was in when the new version was installed over it? Was it part-way through onboarding? Or not even past the welcome page?

**Comment by shankari** [comment]
> Can you clarify what state the old install was in when the new version was installed over it? Was it part-way through onboarding? Or not even past the welcome page?

I believe it was not past the welcome page. Especially on iOS, I had just launched the iOS13 emulator to make sure that there was not WSOD, to ensure that we weren't missing issues in older versions as reported by Anat.

I would suggest the following checks. If all of them work, I am fine to close this issue.

1. Use the devapp with the current master branch
2. Switch to the `onboarding...` branch
3. Verify that the page is rendered correctly

Repeat steps 1-3 above after entering the opcode

1. Install the current production version of the app from the release binaries in the NREL internal repo
1. Install the new staging version over it by using the release binaries from the same location
1. Verify that the welcome page is updated

Repeat steps 1-3 above after entering the opcode

**Comment by JGreenlee** [comment]
Further changes will be needed. It appears that "logout" on the old version of the app cleared the config, but it did not clear the "INTRO_DONE" or "DATA_COLLECTION_CONSENTED_PROTOCOL" markers from local storage.

If the new version of the app encounters a blank config, but "INTRO_DONE" or "DATA_COLLECTION_CONSENTED_PROTOCOL" are still true, it will load in a broken state. The new version of the app will need a backwards-compat check that clears the the localstorage if there is no config.

**Comment by JGreenlee** [comment]
> 1. Use the devapp with the current master branch
> 2. Switch to the `onboarding...` branch
> 3. Verify that the page is rendered correctly
> 
> Repeat steps 1-3 above after entering the opcode

I did this part and was able to switch back-and-forth between `master` and my newest branch without issues. I tested this for both i) before starting onboarding and ii) in the middle of onboarding.

> * Install the current production version of the app from the release binaries in the NREL internal repo
> * Install the new staging version over it by using the release binaries from the same location
> * Verify that the welcome page is updated

The current staging release doesn't have the fixes. To see that it works, wouldn't I need to build the app with the fixes included?
Would it be easier to put out a new staging release that includes the fixes, and test this on real phones?

**Comment by shankari** [comment]
Fair point. Given that it is not reproducible in the devapp, I will consider this "tentatively fixed" and push out an RC today (by tonight at the latest)

**Comment by JGreenlee** [comment]
To clarify, I have a fix for:

> Further changes will be needed. It appears that "logout" on the old version of the app cleared the config, but it did not clear the "INTRO_DONE" or "DATA_COLLECTION_CONSENTED_PROTOCOL" markers from local storage.
> 
> If the new version of the app encounters a blank config, but "INTRO_DONE" or "DATA_COLLECTION_CONSENTED_PROTOCOL" are still true, it will load in a broken state. The new version of the app will need a backwards-compat check that clears the the localstorage if there is no config.

at https://github.com/e-mission/e-mission-phone/pull/1056 and it is ready to also be included in the RC, but it is uncertain whether it fixes the "undefined undefined undefined" issue.

**Comment by shankari** [comment]
@JGreenlee  and @Abby-Wheelis were not able to reproduce this
